perf(language_server): pass file content as a referenced `String`

Sysix opened the pull request:
> This PR optimizes performance by passing file content as a reference (&String) instead of by value (String) throughout the language server's diagnostic and formatting functions. This avoids unnecessary string clones when passing file content through the call chain.

camc314 reviewed (approved):
why `&String` vs `&str`

camc314 commented:
### Merge activity

* **Nov 12, 9:22 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 12, 9:22 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
perf(language_server): pass file content as a referenced `String` (#15568)

> This PR optimizes performance by passing file content as a reference (&String) instead of by value (String) throughout the language server's diagnostic and formatting functions. This avoids unnecessary string clones when passing file content through the call chain.
